### PR TITLE
fix: passwordless authentication fixes and improvements

### LINKED ISSUE #79: User is deleted if Twilio message fails on sms passwordless login
If Twilio errors when attempt to sign in via sms passwordless, the user will be deleted whether it was newly inserted or not

```    
try {
  await twilioClient.messages.create({
    body: `Your code is ${otp}`,
    messagingServiceSid: ENV.AUTH_SMS_TWILIO_MESSAGING_SERVICE_ID,
    to: phoneNumber,
  });
} catch (error) {
  // delete user that was inserted because we were not able to send the SMS
  await gqlSdk.deleteUser({
    userId,
  });

  return res.boom.internal('Error sending SMS');
}
```

### LINKED ISSUE #99: SMS passwordless authentication does check if the user is active or not

### PR DESCRIPTION
**Pull request by L3K0V**
- Add support for Twilio Verify based on the SID use messing or verify APIs
- Closes #79, 
- Closes #99

### CODE REVIEW

**Comment by L3K0V** [comment]
@elitan, @plmercereau if you think is okay while I'm on the twilio topic I can also introduce fixes for https://github.com/nhost/hasura-auth/issues/99, https://github.com/nhost/hasura-auth/issues/79 and maybe https://github.com/nhost/hasura-auth/issues/135

**Comment by plmercereau** [comment]
Hello Aden, yes, definitely!

**Comment by L3K0V** [comment]
@plmercereau can you give some ideas and help about #79. Do you know what should be the expected behaviour - is simply not deleting the user enough?

File: src/routes/signin/passwordless/email.ts [hunk 1]
```diff
@@ -119,5 +119,10 @@ export const signInPasswordlessEmailHandler: RequestHandler<
     },
   });
 
-  return res.send(ReasonPhrases.OK);
+  const signInResponse = await getSignInResponse({
+    userId: user.id,
+    checkMFA: true,
+  });
```
[the remark this answers is not in the document]
Comment (L3K0V) [reply src/routes/signin/passwordless/email.ts:125]: @elitan can you check [this issue](https://github.com/nhost/hasura-auth/issues/135) and suggest how we can proceed? Any ideas? Do I revert these changes, strip the id or something else?

[the remark this answers is not in the document]
Comment (elitan) [reply src/routes/signin/passwordless/email.ts:125]: Yes. I'll follow up in the issue. Would it be possible to break out that issue and code from this PR until we've agreed on what to do?

File: src/routes/signin/passwordless/sms/otp.ts [hunk 2]
```diff
@@ -56,7 +61,33 @@ export const signInOtpHandler: RequestHandler<{}, {}, OtpSmsBody> = async (
     return sendError(res, 'invalid-otp');
   }
 
-  if (!(await bcrypt.compare(otp, user.otpHash))) {
+  if (ENV.AUTH_SMS_PROVIDER !== 'twilio') {
+    throw Error('No sms provider set');
```
[the remark this answers is not in the document]
Comment (L3K0V) [reply src/routes/signin/passwordless/sms/otp.ts:65]: @elitan done, can you please check?

**Review by plmercereau — APPROVED** [review]

**Comment by L3K0V** [comment]
Any progress on the review here?

File: src/routes/signin/passwordless/sms/otp.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (elitan) [reply src/routes/signin/passwordless/sms/otp.ts:65]: Good.

**Review by elitan — APPROVED** [review]

**Pull request merged**